web/src/buddy_list.js

```diff
@@ -662,8 +662,10 @@ export class BuddyList extends BuddyListConf {
         if (new_pos_in_all_users < this.render_count) {
             this.render_count += 1;
             const $li = this.find_li({key: user_id_following_insertion});
-            $li.before(html);
-            this.update_padding();
+            if ($li !== undefined) {
+                $li.before(html);
+                this.update_padding();
+            }
```

Comment:
How did the previous logic work at all? `[].before()` seems like it would crash?

I'm kinda wondering if this `undefined` case is impossible...

Reply:
Yeah that's a good question. It seems like it would have been crashing.

Here's the code where `undefined` is being returned

```js
        // We reference all_user_ids to see if we've rendered
        // it yet.
        const pos = this.all_user_ids.indexOf(user_id);

        if (pos < 0) {
            return undefined;
        }
```

I think that we're only calling `find_li` with items that have already been rendered, though I'm not totally confident of that at this point. We could switch to `assert(pos >= 0)` maybe? I can look into it more to see if that seems reasonable.

Reply:
Actually, is the fact that we've never seen errors for `[].before()` before enough reason to feel confident going with `assert(pos >= 0)`?

Reply:
Yeah assert seems probably correct here.